src/components/SettlementButton.js

```diff
@@ -118,12 +120,28 @@ class SettlementButton extends React.Component {
 
     render() {
         return (
-            <ButtonWithMenu
-                isDisabled={this.props.isDisabled}
-                isLoading={this.props.isLoading}
-                onPress={this.props.onPress}
-                options={this.state.buttonOptions}
-            />
+            <KYCWall
+                onSuccessfulKYC={() => this.props.onPress(CONST.IOU.PAYMENT_TYPE.EXPENSIFY)}
+                enablePaymentsRoute={this.props.enablePaymentsRoute}
+                addBankAccountRoute={this.props.addBankAccountRoute}
+                addDebitCardRoute={this.props.addDebitCardRoute}
```

Comment:
side question that is NAB: is it possible to combine these three routes into one? Like we have one component `EnablePayments` that renders a different view depending on which payment method a user is adding? 

Might be an issue of properly separating concerns and not combining flows like we've talked about before.

Reply:
Anything is possible I guess. :) What are the advantages of doing things that way?

Reply:
Off the bat, I think we'd need to tell that component which view to render somehow with maybe a route param? 

So the route would end up something like `/enable-payments?subpage=activate-wallet` or something? Feels like that would enable one route to be the "parent" route and we wouldn't need to set up so many routes and less boilerplate. But at the same time we'd have kind of an uglier link to that particular screen.

Reply:
> What are the advantages of doing things that way?

> Feels like that would enable one route to be the "parent" route and we wouldn't need to set up so many routes and less boilerplate. 

Yeah, exactly my thoughts. Trade-off is that the parent component would probably have a lot of different logic to handle the payment methods, which would be a bit messier.

Reply:
Probably can be left as an improvement.

Reply:
Yea, another way to look at this is that these things don't really have a _ton_ to do with each other and can all happen in isolation. They are perfectly independent modular parts that are only related in that we _sometimes_ need them to run in a specific sequence. 

So, in all a case where you only need one - you end up having to pull in _all_ of them and still have to define which one you need anyway. Honestly, I haven't found a perfect rule for when to keep things portable or smash them together (so I usually lean towards portability when I can). It's a bit of thinking about how something may be used in the future, not obsessing over it too much, but also trying to get it right the first time (because you might not get a chance to refactor later).

Reply:
Yes, I also a much bigger fan of keeping things separate and portable to begin with because I think it's easier to combine things later in a refactor than pull disparate components out of a single one.